### DISCUSSION TITLE
Using ranges with flexible fields

### DISCUSSION THREAD

**Discussion by JOJ0** [beetbox/beets#4626] [Q&A] [answered]
I have used this before and rather suddenly it seems to have stopped working:

The xtractor plugin (or before that the AcousticBrainz plugin did that), gives me float values for certain flexible attributes.

```
$ ls -f '$mood_acoustic' mood_acoustic:0.9 | head 
0.955127000809
0.971143782139
0.990162730217
0.980634093285
0.999999761581
0.991385340691
0.982989728451
0.9559045434
0.984161794186
0.926920771599#
```

 I'm not sure if they are actually float. Could be strings as well. This could be the problem. The thing is, as said, I'm 100% sure I've used ranges in my smart playlist queries and elsewhere often. They stopped working:

```
$ ls -f '$mood_acoustic' mood_acoustic:0.1..1
```

What did I do with my library to mess this up? Any ideas? Where there any changes to beets recently that could have changed this behaviour? (Even though I'm pretty sure it must be my fault....but I just don't see it...)

I've tested this with the current beets dev version checkout out from master branch.

Thanks for any hints in advance and all the best!

**Comment by arsaboo** [comment]
Just checked the xtractor code and looks like the plugin does not specify the types for each field. You can manually define types using the types plugin and it should work.

**Reply by JOJ0** [reply]
Awesome! Thanks for checking @arsaboo . I will fix this directly in the xtractor  plugin. If you know already around where it should be fixed, please point me there or even open a PR. I have access to the repo and can merge :-)

**Comment by JOJ0** [comment, marked as answer]
I openend an issue directly in the plugin repo. Let's continue discussions there. It's not a beets (core) issue: https://github.com/adamjakab/BeetsPluginXtractor/issues/25
